demo/api/src/products/product.resolver.ts

```diff
@@ -0,0 +1,144 @@
+            populate.push("tags");
+        }
+        if (fields.includes("datasheets")) {
+            populate.push("datasheets");
+        }
+        if (fields.includes("statistics")) {
+            populate.push("statistics");
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const options: FindOptions<Product, any> = { offset, limit, populate };
+
+        if (sort) {
+            options.orderBy = sort.map((sortItem) => {
+                return {
+                    [sortItem.field]: sortItem.direction,
+                };
+            });
+        }
+
+        const [entities, totalCount] = await this.repository.findAndCount(where, options);
+        return new PaginatedProducts(entities, totalCount);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    gqlArgsToMikroOrmQuery({ search, filter }: { search?: string; filter?: any }, repository: EntityRepository<any>): ObjectQuery<any> {
+        const andFilters = [];
+
+        const metadata = repository.getEntityManager().getMetadata().get(repository.getEntityName());
+        if (search) {
+            const crudSearchPropNames = getCrudSearchFieldsFromMetadata(metadata);
+            if (crudSearchPropNames.length == 0) {
+                throw new Error("Entity has no searchable fields");
+            }
+            andFilters.push(searchToMikroOrmQuery(search, crudSearchPropNames));
+        }
+
+        if (filter) {
+            andFilters.push(
+                filtersToMikroOrmQuery(filter, {
+                    metadata,
+                    applyFilter: (
+                        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                        acc: ObjectQuery<any>,
+                        filterValue: StringFilter | NumberFilter | DateTimeFilter | DateFilter | BooleanFilter | ProductManufacturerFilter | unknown,
+                        filterKey: string,
+                    ) => {
+                        if (filterValue instanceof ProductManufacturerFilter) {
+                            if (filterValue.equal !== undefined) {
+                                const query = filtersToMikroOrmQuery(filterValue.equal);
+                                if (Object.keys(query).length > 0) {
+                                    acc[filterKey] = query;
+                                }
+                            }
+                            if (filterValue.notEqual !== undefined) {
+                                const query = filtersToMikroOrmQuery(filterValue.notEqual);
+                                if (Object.keys(query).length > 0) {
+                                    const qb = this.entityManager.createQueryBuilder(Manufacturer).select("id").where(query);
+                                    acc[expr(`${filterKey} not in (${qb.getFormattedQuery()})`)] = true;
+                                }
+                            }
+                        } else if (
```

Comment:
This change would be needed in the library's gqlArgsToMikroOrmQuery, right?

Reply:
it depends on how we want to support dev's doing it this way. api-generator could generate this as well. but yes, checking for generic "`if (filterValue instanceof AdvancedManyToOneFilter)`" and `if (filterValue instanceof AdvancedOneToManyFilter)` could solve this.